lineage/DownwardRecursion.py

```diff
@@ -1,36 +1,52 @@
 '''File holds the code for the downward recursion.'''
 
 import numpy as np
-from .tHMM_utils import max_gen, get_gen, get_daughters
 from .UpwardRecursion import beta_parent_child_func
 
 
 def get_root_gammas(tHMMobj, betas):
-    ''' Need the first gamma terms in the baum welch, which are just the beta values of the root nodes. '''
+    '''need the first gamma terms in the baum welch, which are just the beta values of the root nodes.'''
+    numStates = tHMMobj.numStates
+
     gammas = []
 
-    for num, lineage in enumerate(tHMMobj.population):  # for each lineage in our Population
-        gamma_array = np.zeros((len(lineage), tHMMobj.numStates))
+    for num, lineageObj in enumerate(tHMMobj.X):  # for each lineage in our Population
+        lineage = lineageObj.output_lineage
+        gamma_array = np.zeros((len(lineage), numStates))
+
         gamma_array[0, :] = betas[num][0, :]
         assert np.isclose(np.sum(gamma_array[0]), 1.)
         gammas.append(gamma_array)
 
+        for num, lineageObj in enumerate(tHMMobj.X):  # for each lineage in our Population
+            gammas_0_row_sum = np.sum(gammas[num][0])
+            assert np.isclose(gammas_0_row_sum, 1.)
```

Comment:
so this is failing right, can you make an issue, and we can address it after the merge if it still lingers. might be an initialization thing where depending on the initial thmmm params it can not converge and fail

Reply:
Yea basically for the most part it works. It's just some are summing to 0, most of them are summing to 1. I feel like it might be the leaves? but I'm not sure.

Reply:
it was the leaves last time this happend. and it was cuz their initialization params were too extreme